pkg/signup/signup_service.go

```diff
@@ -238,6 +242,45 @@ func (s *ServiceImpl) GetSignup(userID string) (*Signup, error) {
 	return signupResponse, nil
 }
 
+// PostVerification returns userSignup resource from the host cluster.
+// Returns the usersignup, an http code and an error if present.
+func (s *ServiceImpl) PostVerification(dailyLimit int, responseBody map[string]string, userID, code string) (*crtapi.UserSignup, int, error) {
```

Comment:
The method name is confusing IMO. UpdateWithVerificationCode()?

Also, why you send the entire requestBody (it's not response body btw as it's named now) from the controller to the service. Why not extracting the phone number from the request in the controller and send it to the service as a parameter? It should not return any http statuses either.

Controllers deal with our REST API and HTTP. Services implement business logic and transport/api agnostic.

Reply:
you're right, it shouldn't be called POST and the endpoint shouldn't be a "POST" either. it should really be a PUT, since that's what we're doing... maybe this should be changed?

Reply:
I don't think we need this method at all, the UpdateUserSignup() function should be all we need from this service.

Reply:
Hm, i don't really understand this flow in that case? The verifyCodeHandler calls UpdateUserSignup() - how would this work?

Reply:
this has been fixed here: https://github.com/codeready-toolchain/registration-service/blob/60ca7377aabe427ed8f8e3d642380b1b8396454e/pkg/controller/signup.go#L51